### DISCUSSION TITLE
Symlinks not working on NTFS filesystem shared from Windows Server

### DISCUSSION THREAD

**Discussion by fantomdew** [retronas/discussions#83] [Q&A]
Greetings. i am not sure where to post this as far as a discussion vs an issue so i thought i would start here. I am doing a new install of RetroNas. Everything seems to go fine until i get to install the Mister section. I get errors and they seem to be pointing to missing folders. When i go into roms under Nintendo i noticed that the nes and snes folders are missing. There might be more but those are the two i see right off the bat. This seems to be similar to issues 133 and 137. I see they have been closed so i am not sure if i am doing something wrong or if perhaps something is missing on a fresh install?

thanks

**Comment by sairuk** [comment]
please get a screenshot of the error you are encountering

I have converted this to https://github.com/danmons/retronas/issues/147 please continue the conversation there.

**Comment by sairuk** [comment]
After review of issue #147 it is deemed out of scope for core RetroNAS, user has kindly offered to look into what is possible on their Windows Server

Discussion will remain open as an opportunity for findings

**Comment by fantomdew** [comment]
I just want to put down my observations here as i work on this. I decided to mount an NFS folder from one of my other NAS boxes. I reran the Samba and Mister installs. I still noticed that NES and SNES were missing from the main roms folder but are in the Mister folder now. I added a test folder to the NES under Mister and then searched the drive for the test folder and noticed that Mister\Games\SNES seems to be linked to roms\Ninstendo\wii? I am going to do a fresh start later this evening and see if there is anything left over from other tested and see if it still happens there.
![SNES map](https://user-images.githubusercontent.com/101946302/159182951-983a8984-7563-4d87-8edf-a1db5e7e1c00.PNG)

**Reply by sairuk** [reply]
i've taken a look at this finding and can confirm there is an incorrect entry in the romdir creation which is overriding the expected snes target when the nintendo wii entry is processed.

i've pushed a fix for that now, thanks

**Reply by sairuk** [reply]
i will also add that on a standard deb11 vm (no remote storage) the symlinks for nes/snes are created correctly. I just did fresh reinstall on my test box

**Comment by fantomdew** [comment]
Ok, sounds good,. i did do the update and it seems to be working, I think i also see one of the other issue i was having was the fact i didnt realize that in the rom\nintendo folder the NES and SNES are under famicom and superfamicom. that was throwing me for a loop for awhile. Great product by the way.

**Comment by 40wattrange** [comment]
Posting this from the topic I had opened and was closed as a dup:

Do I need to delete the mister symlink folders first? Just ran the folder update via the retronas.sh file and still having the same issue. Removed files from folder, copied them to SNES via MiSTer share - still went to the Wii folder while SNES/SFAM folder remains empty.

**Reply by sairuk** [reply]
you should not need to remove the path, the symlink target should have been updated providing your RetroNAS install updated correctly, you will either need to restart the UI or choose Update RetroNAS from the Update menu, if you had the UI open then nothing would have updated on its own.

you can check the path in the terminal by running

`la -la <retronas_storage>/mister.games/SNES`

replace retronyou should not need to remove the path, the symlink target should have been updated providing your RetroNAS install updated correctly, you will either need to restart the UI or choose Update RetroNAS from the Update menu, if you had the UI open then nothing would have updated on its own.

you can check the path in the terminal by running

`la -la <retronas_storage>/mister.games/SNES`

replace retronas_storage with whatever your data path is, it should point to superfamicom after the update

Also run from the terminal

`grep wii /opt/retronas/ansible/retronas_systems.yml`

and confirm SNES is not mentioned, if it is your RetroNAS install did not update for some reason, which would be odd

finally if the above are all OK, run Romdirs again and grab `/opt/retronas/log/ansible.log` and look for any errorsas_storage with whatever your data path is, it should point to superfamicom after the update

Also run from the terminal

`grep wii /opt/retronas/ansible/retronas_systems.yml`

and confirm SNES is not mentioned, if it is your RetroNAS install did not update for some reason, which would be odd

finally if the above are all OK, run Romdirs again and grab `/opt/retronas/log/ansible.log` and look for any errors
